packages/shared/lib/contexts/onboarding/actions/findShimmerRewards.ts

```diff
@@ -1,40 +1,111 @@
+            accountGapLimit,
+            addressGapLimit,
+            DEFAULT_SHIMMER_CLAIMING_SYNC_OPTIONS
+        )
+        const boundAccounts = (
+            await Promise.all(
+                accountMetadataList.map(async (accountMetadata) => {
+                    const account = await getBoundAccount(accountMetadata?.index, true, shimmerClaimingProfileManager)
+                    account.meta.alias = Number.isNaN(accountMetadata?.alias)
+                        ? accountMetadata?.alias
+                        : `${localize('general.account')} ${accountMetadata?.index + 1}`
+                    return account
+                })
+            )
+        ).sort(sortAccountsByIndex)
+        const updatedTotalUnclaimedShimmerRewards = await sumTotalUnclaimedRewards(boundAccounts)
+        const wereRewardsFound = updatedTotalUnclaimedShimmerRewards > totalUnclaimedShimmerRewards
+        if (wereRewardsFound) {
+            const boundTwinAccounts = (
+                await Promise.all(
+                    boundAccounts.map(async (boundAccount) => {
+                        const account = await getBoundAccount(boundAccount?.meta?.index, true, profileManager)
+                        account.meta.alias = Number.isNaN(boundAccount?.meta?.alias)
+                            ? boundAccount?.meta?.alias
+                            : `${localize('general.account')} ${boundAccount?.meta?.index + 1}`
+                        return account
+                    })
+                )
+            ).sort(sortAccountsByIndex)
+            for (const [boundAccount, boundTwinAccount] of zip(boundAccounts, boundTwinAccounts)) {
+                const shimmerClaimingAccount = await prepareShimmerClaimingAccount(boundAccount, boundTwinAccount, true)
+                updateShimmerClaimingAccount(shimmerClaimingAccount)
+            }
+
+            const foundRewardsAmount = updatedTotalUnclaimedShimmerRewards - totalUnclaimedShimmerRewards
+            const foundRewardsAmountFormatted = formatTokenAmountBestMatch(
+                foundRewardsAmount,
+                BASE_TOKEN[NetworkProtocol.Shimmer]
+            )
+            showAppNotification({
+                type: 'success',
+                alert: true,
+                message: `Successfully found ${foundRewardsAmountFormatted}`,
+            })
+
+            totalUnclaimedShimmerRewards = updatedTotalUnclaimedShimmerRewards
+
+            accountGapLimit = accountGapLimitIncrement
+            addressGapLimit = addressGapLimitIncrement
+
+            // startAccountIndex = 0
+            // startAddressIndex = 0
+        } else {
+            // startAccountIndex += accountGapLimitIncrement
+            // startAddressIndex += addressGapLimitIncrement
+        }
     } catch (err) {
         console.error(err)
     }
 }
+
+export function setTotalUnclaimedShimmerRewards(initialTotalUnclaimedShimmerRewards: number): void {
```

Comment:
To avoid using a store, this function **MUST** reside within this file as it uses a local variable.